esphome/components/usb_host/__init__.py

```diff
@@ -0,0 +1,55 @@
+import esphome.codegen as cg
+from esphome.components.esp32 import add_idf_sdkconfig_option
+import esphome.config_validation as cv
+from esphome.const import CONF_ID
+from esphome.cpp_types import Component
+
+AUTO_LOAD = ["bytebuffer"]
+CODEOWNERS = ["@clydebarrow"]
+usb_host_ns = cg.esphome_ns.namespace("usb_host")
+USBHost = usb_host_ns.class_("USBHost", Component)
+USBClient = usb_host_ns.class_("USBClient", Component)
+
+CONF_DEVICES = "devices"
+CONF_VID = "vid"
+CONF_PID = "pid"
+
+
+def usb_device_schema(cls=USBClient, vid: int = None, pid: [int] = None) -> cv.Schema:
+    schema = cv.COMPONENT_SCHEMA.extend(
+        {
+            cv.GenerateID(): cv.declare_id(cls),
+        }
+    )
+    if vid:
+        schema = schema.extend({cv.Optional(CONF_VID, default=vid): cv.uint16_t})
+    else:
+        schema = schema.extend({cv.Required(CONF_VID): cv.uint16_t})
+    if pid:
+        schema = schema.extend({cv.Optional(CONF_PID, default=pid): cv.uint16_t})
+    else:
+        schema = schema.extend({cv.Required(CONF_PID): cv.uint16_t})
```

Comment:
Purely visual for codegen so it shows `0x1234` instead of `4660` inside `main.cpp`

```suggestion
        schema = schema.extend({cv.Optional(CONF_VID, default=vid): cv.hex_uint16_t})
    else:
        schema = schema.extend({cv.Required(CONF_VID): cv.hex_uint16_t})
    if pid:
        schema = schema.extend({cv.Optional(CONF_PID, default=pid): cv.hex_uint16_t})
    else:
        schema = schema.extend({cv.Required(CONF_PID): cv.hex_uint16_t})
```